### DISCUSSION TITLE
Line breaks in HTML and PDF

### DISCUSSION THREAD

**Discussion by johnkerl** [quarto-dev/quarto-cli#8478] [Q&A] [answered]
### Description

In a book-format Quarto document, I want the option to occasionally insert a snippet of poetry quotes. I want the line breaks to be respected. I render to HTML and PDF.

Things I've tried:

* Double-spacing: e.g.

```
Line one goes here;

Line two as well
```

Minus: I would like the content to be single-spaced.

* Use `<br>` tags, e.g.

```
Line one goes here;
<br>Line two as well
```

Minus: while this renders beautifully in HTML, in PDF rendering the line breaks are not produced.

* Use pre-formatted, with triple-backtick or indent

Minus: This shows the content in fixed-with font, which I would rather not.

Does anyone have advice?

(I searched in Discussions before posting; my apologies if this is already answered there and I missed it in my searching.)

**Comment by mcanouil** [comment, marked as answer]
I don't understand what you want to do, but that's a markdown question which is not specific to Quarto as GitHub also uses markdown for example.

Line breaks: double space or any number greater than two spaces (or `\`).
Paragraph breaks: one empty line.
See <https://daringfireball.net/projects/markdown/syntax>.

PS: your post is very confusing because it does not really show what you tried and/or the expected results.
Also I do not understand what those "minus" sentences are supposed to mean.

**Comment by lmoresi** [comment]
It's common to want to write a quote that is formatted (font, linespacing) as regular text but has the original line breaks (and not with each line being a separate paragraph). I just needed exactly this to include a song lyric in a class exercise.

I think the closest in quarto's flavour of markdown is the Line Block

| Line Block
|   Spaces and newlines
|   are preserved

<img width="836" alt="Screenshot 2024-04-12 at 10 23 36 AM" src="https://github.com/quarto-dev/quarto-cli/assets/12588963/dc8b6a9a-6b26-4e4b-b75d-745c5362235a">


Which is not rendered by GFM in this comment, but is here: https://quarto.org/docs/authoring/markdown-basics.html#other-blocks

**Reply by johnkerl** [reply]
Nice, thank you @lmoresi !! 😎

**Comment by ute** [comment]
Another option is to use the [`search-replace`](https://github.com/ute/search-replace) extension and define a "macro" for linebreaks that expands into `<br>` when the format is html and into `\newline` when the format is pdf, as shown in the [readme-example](https://github.com/ute/search-replace#example), where the macro was called `+br`.

Disclosure: this is shameless advertisement 😉
